### ISSUE TITLE
Overflow in load_contract_code

### ISSUE THREAD

**Issue by mediremi** [FuelLabs/fuel-vm#191]
`load_contract_code` fails to handle adds with overflow [while validating arguments](https://github.com/FuelLabs/fuel-vm/blob/a5a6759fbb7804bb4375ca299d9f68322a47db4c/src/interpreter/blockchain.rs#L42):

**ssp/length_to_copy_unpadded validation:**
```
thread '<unnamed>' panicked at 'attempt to add with overflow', src/interpreter/blockchain.rs:41:12
stack backtrace:
   0: rust_begin_unwind
             at /rustc/23e21bdd25026e2839ebe946c2a937c1904887d2/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/23e21bdd25026e2839ebe946c2a937c1904887d2/library/core/src/panicking.rs:142:14
   2: core::panicking::panic
             at /rustc/23e21bdd25026e2839ebe946c2a937c1904887d2/library/core/src/panicking.rs:48:5
   3: fuel_vm::interpreter::blockchain::<impl fuel_vm::interpreter::Interpreter<S>>::load_contract_code
   4: fuel_vm::interpreter::executors::instruction::<impl fuel_vm::interpreter::Interpreter<S>>::_instructio
```

**id_addr validation:**
```
thread 'add_with_overflow' panicked at 'attempt to add with overflow', src/interpreter/blockchain.rs:42:16
stack backtrace:
   0: rust_begin_unwind
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/core/src/panicking.rs:142:14
   2: core::panicking::panic
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/core/src/panicking.rs:48:5
   3: fuel_vm::interpreter::blockchain::<impl fuel_vm::interpreter::Interpreter<S>>::load_contract_code
             at ./src/interpreter/blockchain.rs:42:16
   4: fuel_vm::interpreter::executors::instruction::<impl fuel_vm::interpreter::Interpreter<S>>::_instruction
```

Here's a test case (found with the fuzzing script introduced in https://github.com/FuelLabs/fuel-vm/pull/188):

```rust
use fuel_vm::prelude::*;

#[test]
fn ssp_length_to_copy_unpadded_add_with_overflow_in_load_contract_code() {
    let mut client = MemoryClient::default();

    let gas_price = 0;
    let gas_limit = 100_000;
    let maturity = 0;
    let height = 0;
    let params = ConsensusParameters::DEFAULT;

    let mut tx = Transaction::script(
        gas_price,
        gas_limit,
        maturity,
        vec![],
        vec![
                72,
                129,
                90,
                90,
                32,
                255,
                29,
                43,
                72,
                29,
                43,
                255,
                32,
                84,
                49,
                96,
                89,
                255,
                29,
                43,
                50,
                159,
                53,
                91,
                72,
        ],
        vec![],
        vec![],
        vec![],
    )
    .check(height, &params)
    .expect("failed to generate a checked tx");

    client.transact(tx);
}

#[test]
fn id_addr_add_with_overflow_in_load_contract_code() {
    let mut client = MemoryClient::default();

    let gas_price = 0;
    let gas_limit = 100_000;
    let maturity = 0;
    let height = 0;
    let params = ConsensusParameters::DEFAULT;

    let mut tx = Transaction::script(
        gas_price,
        gas_limit,
        maturity,
        vec![],
        vec![
            28,
            74,
            45,
            96,
            50,
            74,
        ],
        vec![],
        vec![],
        vec![],
    )
    .check(height, &params)
    .expect("failed to generate a checked tx");

    client.transact(tx);
}
```

cc @vlopes11 @Voxelot

**Comment by Voxelot** [comment]
related: https://github.com/FuelLabs/fuel-vm/issues/170

**Comment by xgreenx** [comment]
It is fixed in the `fuel-vm 0.33` 
<img width="833" alt="image" src="https://github.com/FuelLabs/fuel-vm/assets/18346821/6f170abf-87de-4878-9b10-559bba5c01c7">
